keyvault/client/src/main/java/com/azure/keyvault/SecretClient.java

```diff
@@ -0,0 +1,540 @@
+         * @throws NullPointerException if {@code credentials} is {@code null}.
+         */
+        public Builder vaultEndPoint(String vaultEndPoint) {
+            Objects.requireNonNull(vaultEndPoint);
+            this.vaultEndPoint = vaultEndPoint;
+            return this;
+        }
+
+        /**
+         * Sets the credentials to use when authenticating HTTP requests.
+         *
+         * @param credentials The credentials to use for authenticating HTTP requests.
+         * @return The updated Builder object.
+         * @throws NullPointerException if {@code credentials} is {@code null}.
+         */
+        public Builder credentials(ServiceClientCredentials credentials) {
+            Objects.requireNonNull(credentials);
+            this.credentials = credentials;
+            return this;
+        }
+
+        /**
+         * Sets the logging level for HTTP requests and responses.
+         *
+         * @param logLevel The amount of logging output when sending and receiving HTTP requests/responses.
+         * @return The updated Builder object.
+         */
+        public Builder httpLogDetailLevel(HttpLogDetailLevel logLevel) {
+            httpLogDetailLevel = logLevel;
+            return this;
+        }
+
+        /**
+         * Adds a policy to the set of existing policies that are executed after
+         * {@link com.azure.keyvault.SecretClient} required policies.
+         *
+         * @param policy The retry policy for service requests.
+         * @return The updated Builder object.
+         * @throws NullPointerException if {@code policy} is {@code null}.
+         */
+        public Builder addPolicy(HttpPipelinePolicy policy) {
+            Objects.requireNonNull(policy);
+            policies.add(policy);
+            return this;
+        }
+
+        /**
+         * Sets the HTTP client to use for sending and receiving requests to and from the service.
+         *
+         * @param client The HTTP client to use for requests.
+         * @return The updated Builder object.
+         * @throws NullPointerException if {@code client} is {@code null}.
+         */
+        public Builder httpClient(HttpClient client) {
+            this.httpClient = client;
+            return this;
+        }
+    }
+
+    public static List<HttpPipelinePolicy> getDefaultPolicies(ServiceClientCredentials credentials, String vaultUrl) {
+        //final ApplicationConfigCredentials credentials = new ApplicationConfigCredentials(connectionString);
+        // Closest to API goes first, closest to wire goes last.
+        final List<HttpPipelinePolicy> policies = new ArrayList<>();
+
+        policies.add(new UserAgentPolicy(String.format("Azure-SDK-For-Java/%s (%s)", SDK_NAME, SDK_VERSION)));
+        policies.add(new RequestIdPolicy());
+        policies.add(new RetryPolicy());
+        policies.add(new KeyvaultCredentialsPolicy(credentials, vaultUrl));
+
+        return policies;
+    }
```

Comment:
If I recall correctly, @conniey was going to remove this API.

Reply:
It is removed in the current state of the branch.

Reply:
Removed it.
After introducing Builder pattern, this is not needed.